exercise/models.py

```diff
@@ -118,40 +127,38 @@ def __str__(self):
     length_minutes = models.IntegerField()
     demographics = models.ManyToManyField(DemographicsInfo, blank=True)
     emails = models.ManyToManyField(ExerciseEmail, blank=True)
-    email_reveal_times = PickledObjectField(null=True)
 
     def set_email_reveal_times(self):
-        # generate email reveal times - these are unique per exercise and are stored in seconds
-        # TODO: (TCKT-1235) minimum amount of emails that should have reveal_time 0
-        # in case the 10% of emails is less than 1 email
-        reveal_times = []
-
-        emails = self.emails.all()
-        if not emails:
+        if not self.emails.all():
             return
+        # Set some emails based on a threshold to zero time initially.
+        threshold_emails = int(ceil((self.emails.count() * settings.REVEAL_TIME_ZERO_THRESHOLD)))
 
-        for email in emails:
-            reveal_times.append(
-                {
-                    'email_id': email.id,
-                    'reveal_time':  randint(0, self.length_minutes * 60)
-                }
-            )
+        while threshold_emails > 0:
+            email = self.emails.all()[randint(0, self.emails.count()-1)]
+            ExerciseEmailRevealTime.objects.get(exercise_id=self.id, email_id=email.id).set_reveal_time(0)
+            threshold_emails -= 1
 
-        # Make 10% of emails appear in inbox at the beginning of exercise
-        received_emails_count = int(emails.count() * 0.1)
-        if received_emails_count >= 1:
-            updated_reveal_times = []
+        # Remaining emails without a set time to be set to a random time
+        for email in self.emails.all():
+            # This creates a random distribution that tends to drift towards the beginning of the exercise.
+            rand_time = int(floor(abs(random() - random()) * (1 + self.length_minutes * 60)))
+            ExerciseEmailRevealTime.objects.get(exercise_id=self.id, email_id=email.id).set_reveal_time(rand_time)
```

Comment:
Just thinking about this again and it'd be fine for now because you only set `reveal_time` if that wasn't set before.
However, this logic might change in the near future so I would still recommend doing what I originally commented.

Reply:
Addressed above.